File: dataset_reduzido/avatar/Python/Code/codeforces_462_B.py
n , k = map ( int , input ( ).split ( ) )
s = input ( )
count = [ 0 ] * 26
for c in s :
    count [ ord ( c ) - ord ( "A" ) ] += 1
count.sort ( reverse = True )
res = 0
for i in range ( 26 ) :
    if count [ i ] >= k :
        res += k * k
        print ( res )
        exit ( )
    k -= count [ i ]
    res += count [ i ] ** 2
print ( res )

File: output/LLaMa/avatar/Python/Java/codeforces_462_B.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int k = scanner.nextInt();
        String s = scanner.next();
        int[] count = new int[26];

        for (char c : s.toCharArray()) {
            if (Character.isUpperCase(c)) {
                count[c - 'A']++;
            }
        }

        Arrays.sort(count);
        int res = 0;

        for (int i = 25; i >= 0; i--) {
            if (count[i] >= k) {
                res += k * k;
                System.out.println(res);
                System.exit(0);
            }
            k -= count[i];
            res += count[i] * count[i];
        }

        System.out.println(res);
    }
}